src/Service/Resolvers/IQueryExecutor.cs

```diff
@@ -69,16 +69,16 @@ public interface IQueryExecutor
                 List<string>? args = null);
 
         /// <summary>
-        /// Extracts first result set and returns it immediately if it has > 0 rows.
-        /// If no rows, tries to get the subsequent result set if any.
-        /// Throws an exception if the second result is null as well.
+        /// Extracts the result set corresponding to the operation (update/insert) being executed.
+        /// For PgSql,MySql, returns the first result set (among the two for update/insert) having non-zero affected rows.
+        /// For MsSql, returns the only result set having non-zero affected rows which corresponds to either update/insert operation.
         /// </summary>
         /// <param name="dbDataReader">A DbDataReader.</param>
         /// <param name="args">The arguments to this handler - args[0] = primary key in pretty format, args[1] = entity name.</param>
         /// <returns>Single row read from DbDataReader.
         /// If the first result set is being returned, DbResultSet.ResultProperties dictionary has
-        /// the property "IsFirstResultSet" set to true.</returns>
-        public Task<DbResultSet> GetMultipleResultSetsIfAnyAsync(
+        /// the property "IsUpdateResultSet" set to true.</returns>
+        public abstract Task<DbResultSet> GetMultipleResultSetsIfAnyAsync(
```

Comment:
>abstract [](http://example.com/codeflow?start=15&length=8)

Redundant/Option to use `abstract` here on Interface method declaration.
> Today, instance members in interfaces are implicitly abstract (or virtual if they have a default implementation), but can optionally have an abstract (or virtual) modifier. Non-virtual instance members must be explicitly marked as sealed. [https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/static-abstracts-in-interfaces#todays-rules](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/static-abstracts-in-interfaces#todays-rules)

You can still map this interface method to onto a virtual method in a class like you do for QueryExecutor to allow your override implementation in MsSqlQueryExecutor.

> [https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/interfaces#1866-interface-implementation-inheritance](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/interfaces#1866-interface-implementation-inheritance)